src/sentry/plugins/providers/integration_repository.py

```diff
@@ -82,18 +82,34 @@ def create_repository(
     ):
         result = self.build_repository_config(organization=organization, data=repo_config)
 
+        integration_id = result.get("integration_id")
+        external_id = result.get("external_id")
+
         repo_update_params = {
             "external_id": result.get("external_id"),
             "url": result.get("url"),
             "config": result.get("config") or {},
             "provider": self.id,
-            "integration_id": result.get("integration_id"),
+            "integration_id": integration_id,
         }
 
-        # first check if there is a repository without an integration that matches
+        # first check if there is an existing repository with an integration that matches
+        existing_repo = Repository.objects.filter(
```

Comment:
We should also filter for HIDDEN, just in case we start using another status later.